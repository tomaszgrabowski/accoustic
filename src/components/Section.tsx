import React from 'react';

interface IProps {
    data: string;
}

const Section = ( props: IProps ) => {
    return (
        <section dangerouslySetInnerHTML={ { __html: props.data } }/>
    );
};

export default Section;
