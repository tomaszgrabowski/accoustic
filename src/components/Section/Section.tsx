import React from 'react';
import './Section.css';

interface IProps {
    data: string;
}

const Section = ( props: IProps ) => {
    return (
        <section dangerouslySetInnerHTML={ { __html: props.data } }/>
    );
};

export default Section;
