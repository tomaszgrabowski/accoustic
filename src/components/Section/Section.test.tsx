import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Section from './Section';

describe( 'Section', () => {
    test( 'given data when render then show section', () => {
        const data = `<p>test</p>`;
        const wrapper = shallow( <Section data={ data }/> );
        expect( wrapper.exists( 'section' ) ).toBeTruthy();
    } );
    
    test( 'given Section component when render then renders properly', () => {
        const wrapper = renderer.create( <Section data={ 'test' }/> ).toJSON();
        expect( wrapper ).toMatchSnapshot();
    } );
} );
