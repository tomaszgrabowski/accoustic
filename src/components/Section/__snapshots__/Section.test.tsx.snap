// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section given Section component when render then renders properly 1`] = `
<section
  dangerouslySetInnerHTML={
    Object {
      "__html": "test",
    }
  }
/>
`;
